### ISSUE TITLE
TabView: Accessibility Update

### ISSUE THREAD

**Issue by jubairmr9** [primefaces/primefaces#11201]
### Description

I want to update the accessibility of the `TabView` component.

Going off the first example in the demo app, from this snippet:

```
<p:tabView>
   <f:facet name="actions">
      Global actions
   </f:facet>
   <p:tab title="Header I">
      <f:facet name="actions">
         Options I
      </f:facet>
      <p class="m-0">
         Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
         quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
         dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
   </p:tab>
   <p:tab title="Header II">
      <f:facet name="actions">
         Options II
      </f:facet>
      <p class="m-0">
         Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab
         illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut
         odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Consectetur, adipisci velit, sed quia non numquam eius modi.
      </p>
   </p:tab>
   <p:tab title="Header III">
      <p class="m-0">At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores
         et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga.
         Et harum quidem rerum facilis est et expedita distinctio. Nam libero tempore, cum soluta nobis est eligendi optio cumque nihil impedit quo minus.
      </p>
   </p:tab>
</p:tabView>
```

We get:
```
<div id="form:j_idt286" class="ui-tabs ui-widget ui-widget-content ui-corner-all ui-hidden-container ui-tabs-top" data-widget="widget_form_j_idt286">
   <ul class="ui-tabs-nav ui-helper-reset ui-widget-header ui-corner-all" role="tablist">
      <li class="ui-tabs-header ui-state-default ui-tabs-selected ui-state-active ui-corner-top" role="tab" aria-expanded="true" aria-selected="true" data-index="0" tabindex="0">
         <a href="#form:j_idt286:j_idt288" tabindex="-1">Header I</a>
      </li>
      <li class="ui-tabs-actions" aria-hidden="false">
         Options I
      </li>
      <li class="ui-tabs-header ui-state-default ui-corner-top" role="tab" aria-expanded="false" aria-selected="false" data-index="1" tabindex="0">
         <a href="#form:j_idt286:j_idt291" tabindex="-1">Header II</a>
      </li>
      <li class="ui-tabs-actions" aria-hidden="true">
         Options II
      </li>
      <li class="ui-tabs-header ui-state-default ui-corner-top" role="tab" aria-expanded="false" aria-selected="false" data-index="2" tabindex="0">
         <a href="#form:j_idt286:j_idt294" tabindex="-1">Header III</a>
      </li>
      <li class="ui-tabs-actions ui-tabs-actions-global" aria-hidden="true">
         Global actions
      </li>
   </ul>
   <div class="ui-tabs-panels">
      <div id="form:j_idt286:j_idt288" class="ui-tabs-panel ui-widget-content ui-corner-bottom" role="tabpanel" aria-hidden="false" data-index="0">
         <p class="m-0">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, 
            quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum 
            dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
         </p>
      </div>
      <div id="form:j_idt286:j_idt291" class="ui-tabs-panel ui-widget-content ui-corner-bottom ui-helper-hidden" role="tabpanel" aria-hidden="true" data-index="1">
         <p class="m-0">
            Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab 
            illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut 
            odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Consectetur, adipisci velit, sed quia non numquam eius modi.
         </p>
      </div>
      <div id="form:j_idt286:j_idt294" class="ui-tabs-panel ui-widget-content ui-corner-bottom ui-helper-hidden" role="tabpanel" aria-hidden="true" data-index="2">
         <p class="m-0">At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores 
            et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga. 
            Et harum quidem rerum facilis est et expedita distinctio. Nam libero tempore, cum soluta nobis est eligendi optio cumque nihil impedit quo minus.
         </p>
      </div>
   </div>
   <input id="form:j_idt286_activeIndex" name="form:j_idt286_activeIndex" type="hidden" autocomplete="off" value="0">
</div>
```

### Describe the solution you would like

Here are the structural changes I'd like to propose to make it more accessible:
- Move all the aria attributes from the `li` elements and shift them to the `a` element
- In the `li` element we won't set the `tabindex="-1"` attribute because having them, we cannot focus on them by the keyboard when navigating with the Tab and Shift + Tab keys.
- Add the attribute `role="presentation"` to the `li` element after
- In the `a` element, add functionality that if `aria-selected="true"` then `tabindex="0"`, otherwise `tabindex="-1"`
- Generate an `id` dynamically for the `a` element if not provided (ex - `id="tab-1"`), and this value will be referred to in the attribute `aria-labelledby="tab-1"` of the corresponding div with the attribute `role="tabpanel"`
- Generate an `id` dynamically for the div element if not provided with the attribute `role="tabpanel"` (ex - `id="tabpanel-1"`), and this value will be referred to in the attribute `aria-controls="tabpanel-1"` of the corresponding `a` element with the attribute `role="tab"`
- Add a `tabindex="0"` to each `div` element with attribute `role="tabpanel"` to make the component easier for inexperienced keyboard-only users to navigate

Here are the functional ones, and since this will be a manually activated tab:
- <kbd>Left</kbd> = highlight the previous tab without activating it.
- <kbd>Right</kbd> = highlight the next tab without activating it.
- <kbd>Enter</kbd> or <kbd>Space</kbd> = activate the currently highlighted tab.
- <kbd>Home</kbd> / <kbd>PageUp</kbd>_(optional)_ = highlight the first tab without activating it.
- <kbd>End</kbd> /  <kbd>PageDown</kbd> _(optional)_ = highlight the last tab without activating it.
